File: Edinburgh College/JS/5. Functions/activity2.js
// Function to calculate factorial
function factorial(number) {
    let totalFactorial = 0;
    // Loop and add from 1 until the user input number
    for (let i = 1; i <= number; i++) {
        totalFactorial += i;
    }
    return totalFactorial;  // Return factorial for print
}

// Get input number
while (true) {
    const userNumber = Number(prompt("Enter a number: "));
    let toFactor = 0;
    if (userNumber < 0) {
        console.log("Cannot be a negative number.") // Loop again
    } else if (userNumber == 0) {
        toFactor = 1;
        console.log(factorial(toFactor));   // Print factorial of 1
        break;
    } else {
        toFactor = userNumber;
        console.log(factorial(toFactor));   // Calculate and print factorial
        break;
    }
}

File: Edinburgh College/Python/5. Python Functions/activity2.py
# Factorial from function

# Function to get factorial
def factorial(number):
    # Initialise total variable
    total = 1
    # Check if number is 0
    if number == 0:
        return 1
    # Check if number is less than 0
    elif number < 0:
        return "Invalid number"
    # Loop from 1 to user number
    for integer in range(1, number + 1):
        # Multiply current number per iteration
        total *= integer
    return total


# Get number from user
user_number = int(input("Enter number: "))
# Get the returned value from function with user input parameter
factorial_number = factorial(user_number)
# Check if the value returned is string
if type(factorial_number) is str:
    print(factorial_number)
# If not, then print factorial as normal
else:
    print(f"Factorial of {user_number} is {factorial_number}")